client/cpu_sched.cpp

```diff
@@ -146,6 +146,8 @@ struct PROC_RESOURCES {
                 atp->needs_shmem = false;
             }
         }
+        if (rp->rr_sim_misses_deadline) return false;
+            // Don't schedule if the result is simulated to miss its deadline
```

Comment:
So, such tasks that could possibly not meet the deadline will never have a chance to run?
I think this is not nice behavior, especially for those projects who has sometimes some very small tasks after the big one.